### PR TITLE
Add basic PyTorch inference and training example.

### PR DESCRIPTION
**Pull request by dellis23**

### CODE REVIEW

**Comment by silvasean** [comment]
This looks great, thanks Dan.

**Comment by dellis23** [comment]
Friendly ping @ScottTodd for review.

File: docs/website/docs/getting-started/pytorch.md [hunk 1]
```diff
@@ -0,0 +1,100 @@
+# PyTorch Integration
+
+IREE supports compiling and running PyTorch programs represented as
+`nn.Module` [classes](https://pytorch.org/docs/stable/generated/torch.nn.Module.html) as well as models defined using [`functorch`](https://pytorch.org/functorch/).
+
+## Prerequisites
+
+Install IREE pip packages, either from pip or by
```
Comment (jpienaar) [inline docs/website/docs/getting-started/pytorch.md:8]: either from PyPi, Github nightly builds or by ... ? (I don't know if we have a good link to using the nightly builds, pip installing from github, we could refer to email Scott send to iree-discuss@googlegroup.com as worst case)

File: docs/website/docs/getting-started/pytorch.md [hunk 2]
```diff
@@ -0,0 +1,100 @@
...
+[building from source](../building-from-source/python-bindings-and-importers.md):
+
+```shell
+pip install \
+  iree-compiler \
+  iree-runtime \
+  iree-tools-tf
```
Comment (jpienaar) [inline docs/website/docs/getting-started/pytorch.md:15]: Is iree-tools-tf needed?

File: docs/website/docs/getting-started/pytorch.md [hunk 3]
```diff
@@ -0,0 +1,100 @@
...
+### Compile the MLIR to IREE VM flatbuffer
+
+Next, we compile the resulting MLIR to IREE VM flatbuffer: 
+
+```python
+iree_backend = "llvm-cpu"
+iree_vmfb = iree_torch.compile_to_vmfb(mlir, iree_backend)
+```
+
+Here we have a choice of backend we want to target.  See [IREE Deployment Configurations](https://iree-org.github.io/iree/deployment-configurations/) for a full list of targets.
```
Comment (jpienaar) [inline docs/website/docs/getting-started/pytorch.md:66]: Nit: no double space after period.

**Review by jpienaar — APPROVED** [review]
Thanks! I think this is a good entry point and colabs showing end-to-end (post issues unrelated to the colabs that you are addressing) will be good entry spot.

File: docs/website/docs/getting-started/pytorch.md (hunk 1 above)
Comment (dellis23) [reply docs/website/docs/getting-started/pytorch.md:8]: Hmm, sending an email seems a bit much to ask of the users for something like that.  I think we should probably develop some documentation on how to access them.  I can't find anything existing on Google, at least.  I'll leave it out for now until we have a guide on nightly builds.

File: docs/website/docs/getting-started/pytorch.md (hunk 2 above)
Comment (dellis23) [reply docs/website/docs/getting-started/pytorch.md]: Whoops, a small bit of leftover copy pasting!  Removed.

File: docs/website/docs/getting-started/pytorch.md (hunk 3 above)
Comment (dellis23) [reply docs/website/docs/getting-started/pytorch.md]: In theory markdown should be readable in plaintext format, where it's often done in monospace fonts.  The double space after the period helps distinguish between end of sentence vs splitting words in those cases.  I think it's a bit of a relic from the typewriter days, but for the case like this where we expect it to potentially still be viewed monospace I think it makes sense.

That being said, if the other markdown files aren't following suit, I should just be consistent.  I've updated it.

**Comment by ScottTodd** [comment]
> Friendly ping @ScottTodd for review.
> 
> Edit: Whoops, didn't realize you were OOO. Adding Jacques.

No problem, this is great :). I sent a small PR with a few tweaks: https://github.com/iree-org/iree/pull/10886

**Pull request merged**